## Problem #125: Palindromic Sums
[Link to Problem](https://projecteuler.net/problem=125)

### Approach

We can just bruteforce: we fix the first number and we keep increasing the series of consecutive numbers.

In [6]:
limit = 10 ** 8

unique = set()

for a in range(1, int(limit ** 0.5) + 1):
    sum = a ** 2 + (a + 1) ** 2
    b = a + 1
    while sum < limit:
        if str(sum) == str(sum)[::-1]:
            unique.add(sum)
        b += 1
        sum += b ** 2

sum = 0
for x in unique:
    sum += x

print(sum)

2906969179


###### Result: **2906969179** | Execution time: 0s

### Complexity analysis

The first `for` runs in $O(\sqrt{limit})$. The second `for` runs in $O(\sqrt[3]{limit})$ (think of the formula for the sum of consecutive squares starting from 1). Therefore the time complexity is $O(limit^{\frac{5}{6}})$

##### Tags: #bruteforce, #squares